packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/ClientEventService.java

```diff
@@ -125,42 +144,38 @@ public <T> void sendToUser(final ClientEvent<T> event, final String userId) {
 	 * @param message the message to send
 	 * @param channel the channel to send the message on
 	 */
-	// TODO: use anonymous queues, currently this wont behave correctly with
-	// multiple hmi-server instances. Issue #2679
-	@RabbitListener(queues = "${terarium.client-all-user-event-queue}", concurrency = "1")
+	@RabbitListener(
+			bindings =
+					@QueueBinding(
+							value =
+									@Queue(
+											value =
+													"#{T(java.util.UUID).randomUUID().toString().concat('-all-users-event-queue')}",
+											durable = "false",
+											autoDelete = "true"),
+							exchange = @Exchange(value = CLIENT_ALL_USERS_EVENT_EXCHANGE, type = "fanout")))
 	void onSendToAllUsersEvent(final Message message, final Channel channel) {
 		final JsonNode messageJson = decodeMessage(message, JsonNode.class);
 		if (messageJson == null) {
 			return;
 		}
-		synchronized (userIdToEmitter) {
+		synchronized (userIdToEmitters) {
 			// Send the message to each user connected and remove disconnected users
-			userIdToEmitter.forEach((userId, emitterList) -> {
-				send(messageJson, emitterList, userId);
-			});
-		}
-	}
-
-	private void send(final Object message, final List<SseEmitter> emitterList, final String userId) {
-		final List<SseEmitter> emittersToRemove = new ArrayList<>();
-		emitterList.forEach((emitter) -> {
-			try {
-				emitter.send(message);
-			} catch (final IllegalStateException | ClientAbortException e) {
-				log.warn("Error sending all users message to user {}. User likely disconnected", userId);
-				emittersToRemove.add(emitter);
-			} catch (final IOException e) {
-				log.error("Error sending all users message to user {}", userId, e);
-			}
-		});
-
-		synchronized (userIdToEmitter) {
-			emittersToRemove.forEach((emitter) -> {
-				userIdToEmitter.get(userId).remove(emitter);
-				if (userIdToEmitter.get(userId).size() == 0) {
-					userIdToEmitter.remove(userId);
+			userIdToEmitters.forEach((userId, emitterList) -> {
```

Comment:
To be consistent this should be using the synchronize

Reply:
fixed